src/refiners/foundationals/latent_diffusion/stable_diffusion_1/model.py

```diff
@@ -143,3 +143,27 @@ def set_inpainting_conditions(
         self.target_image_latents = self.lda.encode(x=masked_init_image)
 
         return self.mask_latents, self.target_image_latents
+
+    def compute_self_attention_guidance(self, x, noise, step: int, *, clip_text_embedding, **kwargs):
+        sag = self._find_sag_adapter()
+        assert sag is not None
+
+        degraded_latents = sag.compute_degraded_latents(
+            scheduler=self.scheduler,
+            latents=x,
+            noise=noise,
+            step=step,
+            classifier_free_guidance=True,
+        )
+
+        negative_embedding, _ = clip_text_embedding.chunk(2)
+        timestep = self.scheduler.timesteps[step].unsqueeze(dim=0)
+        self.set_unet_context(timestep=timestep, clip_text_embedding=negative_embedding, **kwargs)
+        x = torch.cat(
+            tensors=(degraded_latents, self.mask_latents, self.target_image_latents),
+            dim=1,
+        )
+        result = self.unet(x)
+        degraded_noise = result[:, :4, :, :]
```

Comment:
This is not needed: the predicted noise has just 4 channels

(only the first conv of the unet is impacted - 4 -> 9 channels - the rest remains identical)